autopilot/contractor.go

```diff
@@ -825,35 +816,13 @@ func contractMapBool(contracts []types.FileContractID) map[types.FileContractID]
 	return cmap
 }
 
-func calculateHostCollateral(cfg api.AutopilotConfig, settings rhpv2.HostSettings, renterFunds, txnFee types.Currency) (types.Currency, error) {
-	// check underflow
-	if settings.ContractPrice.Add(txnFee).Cmp(renterFunds) > 0 {
-		return types.ZeroCurrency, errors.New("contract price + fees exceeds funding")
-	}
-
-	// avoid division by zero
-	if settings.StoragePrice.IsZero() {
-		settings.StoragePrice = types.NewCurrency64(1)
-	}
-
-	// calculate the host collateral
-	renterPayout := renterFunds.Sub(settings.ContractPrice).Sub(txnFee)
-	maxStorage := renterPayout.Div(settings.StoragePrice)
+func calculateHostCollateral(cfg api.AutopilotConfig, settings rhpv2.HostSettings) types.Currency {
 	expectedStorage := cfg.Contracts.Storage / cfg.Contracts.Hosts
-	hostCollateral := maxStorage.Mul(settings.Collateral)
-
-	// don't add more than 5x the collateral for the expected storage to save on fees
-	maxRenterCollateral := settings.Collateral.Mul64(cfg.Contracts.Period).Mul64(expectedStorage).Mul64(5)
-	if hostCollateral.Cmp(maxRenterCollateral) > 0 {
-		hostCollateral = maxRenterCollateral
-	}
-
-	// don't add more collateral than the host would allow
+	hostCollateral := settings.Collateral.Mul64(expectedStorage)
```

Comment:
2 things:

1. Iirc collateral is expressed as Hastings / byte / block so you need to multiply by the contract duration here. 
2. I'd rename `calculateHostCollateral` to `calculateInitialHostCollateral` or something else that indicates this should only be used for the initial formation of a contract.